Separate reduce functions from math

xiaomengy opened the pull request:
Summary:
Separate CPU reduce functions from math

i-am-not-moving-c2-to-c10

Differential Revision: D13999469

yinghai commented on caffe2/utils/math_test.cc:
```diff
@@ -426,253 +426,6 @@ TEST(MathTest, FloatToHalfConversion) {
 
 namespace {
 
-class ReduceTensorTest : public testing::Test {
```
Where do these tests go?

BIT-silence replied:
This part is actually covered by reduce_ops_test. So there is no need to keep such cc_tests.

houseroad commented on caffe2/utils/math/reduce.h:
```diff
@@ -5,8 +5,85 @@
+// Y = alpha * ReduceMin(X)
+template <typename T, class Context>
+CAFFE2_API void ReduceMin(
+    const int ndim,
```
Explicitly mention that X_dims and Y_dims share the same ndim?

BIT-silence replied:
Done

houseroad commented on caffe2/utils/math/reduce.h:
```diff
@@ -5,8 +5,85 @@
+CAFFE2_API void
+ReduceMax(const int N, const T* X, T* y, Tensor* scratch_ptr, Context* context);
+
+// Y = alpha * ReduceMin(X)
```
thumb for the comment :-)

houseroad commented on caffe2/utils/math/reduce.h:
```diff
@@ -19,6 +96,7 @@ CAFFE2_API void Moments(
     Context* context);
 
 } // namespace math
+
```
is this expected?

BIT-silence replied:
Yes, this } is the end of namespace math.

houseroad reviewed (approved):
Looks good.

xiaomengy committed:
Separate reduce functions from math (#16929)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16929

Separate CPU reduce functions from math

i-am-not-moving-c2-to-c10

Reviewed By: houseroad

Differential Revision: D13999469

fbshipit-source-id: 0525f5762d61118b76172c789545dc058f68bd4b
